[jereeemyyyy] iP

jereeemyyyy opened the pull request:
# Talkie
“You can do anything, but not everything.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Talkie, your personal assistant that is always ready to remember and remind!

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://github.com/jereeemyyyy/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [X] Managing tasks
- [X] Managing deadlines
- [x] Managing events 
- [ ] Improved GUI (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jereeemyyyy committed:
Level-0

jereeemyyyy committed:
Level-0

jereeemyyyy committed:
Level-1

jereeemyyyy committed:
Level-2

jereeemyyyy committed:
Level-3

jereeemyyyy committed:
Revert "Level-3"

This reverts commit a9b8995cce089127785db39e337a73c7dba06cc3.

jereeemyyyy committed:
Level-3

jereeemyyyy committed:
Level-4: Add support for tracking three types of tasks

jereeemyyyy committed:
A-TextUiTesting: Add automated text UI testing

jereeemyyyy committed:
Amended Talkie

jereeemyyyy committed:
Updated A-TextUiTesting test cases

jereeemyyyy committed:
Level-5: Error handling

jereeemyyyy committed:
Updated A-TextUitesting test cases to include new feature

jereeemyyyy committed:
Level-6: Add support for deleting tasks from the list

jereeemyyyy committed:
Added enums for the messages

jereeemyyyy committed:
Level-7: Save functionality

jereeemyyyy committed:
Added data path into gitignore

jereeemyyyy committed:
Level-7: Save functionality

jereeemyyyy committed:
Level-8: Use Date & Time in Java

jereeemyyyy committed:
Updated A-TextUitesting test cases with updated inputs

jereeemyyyy committed:
A-MoreOOP: Abstracted Ui class

jereeemyyyy committed:
A-MoreOOP: Abstracted Storage class

jereeemyyyy committed:
A-MoreOOP: Abstracted TaskList class

jereeemyyyy committed:
A-MoreOOP: Refactored Talkie class

jereeemyyyy committed:
Added text-ui-test/data/

jereeemyyyy committed:
Edited the stringify method

jereeemyyyy committed:
Fixed test inputs

jereeemyyyy committed:
Abstracted the parser to parse commands

jereeemyyyy committed:
Refactored Talkie class

jereeemyyyy committed:
Abstracted each command into separate command classes

jereeemyyyy committed:
Added text-ui-test to gitignore

jereeemyyyy committed:
Abstracted several classes

Extracting out related code as classes increases readibility of the code
base

jereeemyyyy committed:
Add command, components, exception and task packages to arrange code

jereeemyyyy committed:
Refactor codebase with the new packages

jereeemyyyy committed:
Add Data directory to .gitignore

jereeemyyyy committed:
Merge add-gradle-support

jereeemyyyy committed:
Refactor Talkie into the correct directory

jereeemyyyy committed:
Refactor the Ui Class

jereeemyyyy committed:
Edit expected outputs of the script test

jereeemyyyy committed:
A-JUnit: Add JUnit tests for TaskList class

jereeemyyyy committed:
Merge branch 'branch-A-JUnit'

jereeemyyyy committed:
A-JUnit: Add JUnit tests for Todo class

jereeemyyyy committed:
Edit the build.gradle file

jereeemyyyy committed:
Add JavaDoc for classes in command package

jereeemyyyy committed:
Add JavaDoc for classes in components package

jereeemyyyy committed:
Add JavaDoc for classes in exception package

jereeemyyyy committed:
Add JavaDoc for classes in task package

jereeemyyyy committed:
Add JavaDoc for main Talkie class

jereeemyyyy committed:
Add find functionality to Talkie

jereeemyyyy committed:
Update JavaDoc for FindCommand and Ui class

jereeemyyyy committed:
Edit codebase to meet Java Coding Standard

jereeemyyyy committed:
Add checkstyle configs

jereeemyyyy committed:
Edit files to meet Standard Java Coding Standards

jereeemyyyy committed:
Edit return type to String for execute method

jereeemyyyy committed:
Remove horizontal lines from print message

jereeemyyyy committed:
Add JavaFx config to build.gradle

jereeemyyyy committed:
Add JavaFx Gui

jereeemyyyy committed:
Add better gui to chatbot

jereeemyyyy committed:
Edit JavaDoc to classes in task package

jereeemyyyy committed:
Edit JavaDoc to command classes

jereeemyyyy committed:
Edit code to meet Java coding standards

jian7490 commented on src/main/java/talkie/Talkie.java:
```diff
@@ -0,0 +1,97 @@
+ */
+public class Talkie {
+
+    private static String filePath = "./data/talkie.Talkie.txt";
```
This is a CONSTANT which should be named in UPPERCASE

jian7490 commented on src/main/java/talkie/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,85 @@
+        String[] parts = fullCommand.split(" ", 2); // Split into type and the rest of the input
+
+        try {
+            if (parts.length == 2) {
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+
+                String details = parts[1]; // Rest of the input (e.g., description /by deadline details)
+                String[] deadlineParts = details.split("/by ");
+                String description = deadlineParts[0].trim();
+                String by = deadlineParts[1].trim();
+
+                LocalDateTime time = LocalDateTime.parse(by, formatter);
+
+                Task newDeadline = new Deadline(description, time);
+                tasks.addTask(newDeadline);
+                return ui.addMessage(newDeadline, tasks.size());
```
I like how you intentionally use lines to separate different logic chunks!

jian7490 commented on src/main/java/talkie/command/DeleteCommand.java:
```diff
@@ -0,0 +1,94 @@
+     * @throws TalkieInvalidArgumentException  If the provided argument is not a valid integer.
+     */
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage)
+            throws TalkieMissingArgumentException, TalkieNoTaskFoundException, TalkieInvalidArgumentException {
```
Great to see you break long lines to improve readability.

jian7490 commented on src/main/java/talkie/components/Ui.java:
```diff
@@ -0,0 +1,205 @@
+    /**
+     * Displays a welcome message to the user.
+     */
+    public String welcomeMessage() {
+        return MessageType.WELCOME_MESSAGE.getMessage();
+    }
+
+    /**
+     * Displays a goodbye message to the user.
+     */
+    public String byeMessage() {
+        return MessageType.BYE_MESSAGE.getMessage();
+    }
+
```
The names for these two methods are more like nouns instead of verbs. Perhaps welcome() and bye() will be more like method names.

jian7490 reviewed (commented):
The code is clean and great in terms of code readability!

jereeemyyyy committed:
Add various assertions

To check whether the values are initialised properly at various
points in the code.

jereeemyyyy committed:
Merge pull request #1 from jereeemyyyy/branch-A-Assertions

Add various assertions

jereeemyyyy committed:
Refactor DeadlineCommand class

DeadlineCommand execute method is very long and almost unreadable.

Abstract the logic within the execute command so that code is
more readible.

jereeemyyyy committed:
Merge pull request #2 from jereeemyyyy/branch-A-CodeQuality

Refactor DeadlineCommand class

jereeemyyyy committed:
Add Sort command

Allow users to sort a list of task alphabetically.

jereeemyyyy committed:
Refactor relevant classes

Create relevant methods in CommandType, SortCommand, Parser
and Ui classes to work with the new SortCommand.

jereeemyyyy committed:
Merge pull request #3 from jereeemyyyy/branch-BCD-Extension

Merge branch-BCD-Extension

jereeemyyyy committed:
Fix method name in TaskList class

There was a typo in the sortByDescription() method.
Initially it was named sortByDesciption. However, the
intended name was sortByDescription, with a missing r.

jereeemyyyy committed:
Merge pull request #4 from jereeemyyyy/branch-BCD-Extension

Fix method name in TaskList class

jereeemyyyy committed:
Add streams to TaskList class

Use streams for the sortByDescription method in the TaskList class.
This is to improve readibilty and optimization for performance.

jereeemyyyy committed:
Merge pull request #5 from jereeemyyyy/branch-A-Streams

Add streams to TaskList class

jereeemyyyy committed:
Set up a sample workflow

Use the workflow specified by the yml file that was provided to us.

jereeemyyyy committed:
Merge pull request #6 from jereeemyyyy/branch-A-CI

Set up a sample workflow

jereeemyyyy committed:
Add User guide for Talkie

Update the readme to have the user guide for Talkie.
Included a Quick start tutorial and listed out the features
for Talkie for the reference for users.

jereeemyyyy committed:
Merge pull request #8 from jereeemyyyy/branch-A-UserGuide

Add User guide for Talkie

jereeemyyyy committed:
Update UserGuide

Add link breaks inbetween each feature for better readibility

jereeemyyyy committed:
Merge pull request #9 from jereeemyyyy/branch-A-UserGuide

Update UserGuide

jereeemyyyy committed:
Fix syntax errors in readme

Fix some minor spelling and syntax errors in User guide

jereeemyyyy committed:
Merge branch 'branch-A-UserGuide'

jereeemyyyy committed:
Add title and icon to gui

Add title and icon to gui of talkie taskbot for customisation!

jereeemyyyy committed:
Merge pull request #10 from jereeemyyyy/branch-A-BetterGui

Add title and icon to gui

jereeemyyyy committed:
Update snippet picture of Talkie

Change the snippet picture of Talkie to include the title and icon
at the top of the window.

jereeemyyyy committed:
Merge pull request #11 from jereeemyyyy/branch-A-UserGuide

Update snippet picture of Talkie

jereeemyyyy committed:
Update Javadoc for some classes

Fix some Javadoc for Launcher, Talkie, SortCommand and TaskList class

jereeemyyyy committed:
Merge pull request #13 from jereeemyyyy/branch-fix-javadoc

Update Javadoc for some classes
